main.go

```diff
@@ -156,20 +156,28 @@ func calc(stack *stackType, cmd string) error {
 			continue
 		}
 
+		if autoprint {
+			stack.printTop(ops.base, single)
+		}
+
 		// Break after the first iteration if a command is passed.
 		if single {
 			break
 		}
-		if autoprint {
-			stack.printTop(ops.base)
-		}
 	}
 	return nil
 }
 
 func main() {
 	stack := &stackType{}
-	if err := calc(stack, ""); err != nil {
+
+	// treat the cmdline arguments as an expression to be processed
+	expr := ""
```

Comment:
You can replace these lines with:
```
expr := strings.Join(os.Args[1:])
```
or add the `strings.Join` directly to line 180 below.

Reply:
> or add the strings.Join directly to line 180 below.

Done, albeit strings.Join() needed an extra argument (the separator): 
```
    if err := calc(stack, strings.Join(os.Args[1:], " ")); err != nil {
        log.Fatal(err)
    }
```

Tested and working, PR updated.